firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/UserMetadata.java

```diff
@@ -63,6 +60,40 @@ public void setCustomKey(String key, String value) {
     attributes.put(key, value);
   }
 
+  public void setCustomKeys(Map<String, String> keysAndValues) {
+    // Update any existing keys first, then add any additional keys
+    Map<String, String> currentKeys = new HashMap<String, String>();
+    Map<String, String> newKeys = new HashMap<String, String>();
+
+    // Update current keys
+    for (Map.Entry<String, String> entry : keysAndValues.entrySet()) {
+      String key = sanitizeKey(entry.getKey());
+      String value = (entry.getValue() == null) ? "" : sanitizeAttribute(entry.getValue());
+
+      if (attributes.containsKey(key)) {
+        currentKeys.put(key, value);
+      } else {
+        newKeys.put(key, value);
+      }
+    }
+    attributes.putAll(currentKeys);
+
+    // Add new keys if there is space
```

Comment:
Should we at least try to add *some* of them? This is a tricky question since the map will be unordered, but disallowing *any* new values also seems suboptimal.

Reply:
Added an alphabetical sort to add however many we can in a predictable way